modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/AbstractMvTableStorageTest.java

```diff
@@ -311,51 +313,89 @@ public void testMisconfiguredIndices() {
 
     @Test
     public void testStartRebalanceMvPartition() throws Exception {
+        assertThrows(StorageException.class, () -> tableStorage.startRebalanceMvPartition(PARTITION_ID_1));
+
+        UUID hashIndexId = hashIdx.id();
+        UUID sortedIndexId = sortedIdx.id();
+
         MvPartitionStorage partitionStorage = tableStorage.getOrCreateMvPartition(PARTITION_ID);
+        HashIndexStorage hashIndexStorage = tableStorage.getOrCreateHashIndex(PARTITION_ID, hashIndexId);
+        SortedIndexStorage sortedIndexStorage = tableStorage.getOrCreateSortedIndex(PARTITION_ID, sortedIndexId);
+
+        RowId rowId = new RowId(PARTITION_ID);
+
+        TestKey key = new TestKey(0, "0");
+
+        BinaryRow binaryRow = binaryRow(key, new TestValue(1, "1"));
+        IndexRowImpl indexRow = new IndexRowImpl(keyValueBinaryTuple(binaryRow), rowId);
 
         partitionStorage.runConsistently(() -> {
-            partitionStorage.addWriteCommitted(
-                    new RowId(PARTITION_ID),
-                    binaryRow(new TestKey(0, "0"), new TestValue(1, "1")),
-                    clock.now()
-            );
+            partitionStorage.addWriteCommitted(rowId, binaryRow, clock.now());
 
             partitionStorage.lastAppliedIndex(100);
 
             return null;
         });
 
+        hashIndexStorage.put(indexRow);
```

Comment:
Why are you doing it without "runConsistently"? All writes should be inside of the closure.
And index updates should be in the same closure as writes, aren't they? It's irrelevant for this particular test, but as a general rule it's fine, why violating it?

Reply:
Technically it can do this, but you're right, let it be inside `runConsistently`.